provider.go

```diff
@@ -1,184 +1,257 @@
+// Package hetzner implements the [libdns](https://github.com/libdns/libdns) interfaces for the Hetzner Cloud DNS API.
 package hetzner
 
 import (
 	"context"
-	"fmt"
+	"errors"
+	"os"
 	"sync"
 
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
 	"github.com/libdns/libdns"
 )
 
-// Provider implements the libdns interfaces for Hetzner.
+const (
+	applicationName    = "github.com/libdns/hetzner"
+	applicationVersion = "v2"
+
+	envVarName = "LIBDNS_HETZNER_TOKEN"
+)
+
+// Provider implements the libdns interfaces for the Hetzner DNS API.
 type Provider struct {
-	// AuthAPIToken is the Hetzner Auth API token - see https://dns.hetzner.com/api-docs#section/Authentication/Auth-API-Token
-	AuthAPIToken string `json:"auth_api_token"`
+	// APIToken is the Hetzner Cloud API token to use with HTTP requests. If not set, it defaults to the
+	// LIBDNS_HETZNER_TOKEN environment variable.
+	//
+	// See https://docs.hetzner.cloud/reference/cloud#getting-started to learn how to generate a new token.
+	APIToken string `json:"api_token,omitempty"`
 
-	client *Client
+	client *hcloud.ZoneClient
 	once   sync.Once
 }
 
-// New returns a new libdns provider for Hetzner.
-func New(token string) *Provider {
```

Comment:
Keep the New constructor. This is very common in Go libs and provides a way to ask for all required fields instead of relying on the user to understand which fields are actually required. Not so much in this case of course, but it's still a good practice imo.